app/src/main/java/com/stevesoltys/seedvault/settings/ExpertSettingsFragment.kt

```diff
@@ -1,15 +1,29 @@
 package com.stevesoltys.seedvault.settings
 
 import android.os.Bundle
+import androidx.activity.result.contract.ActivityResultContracts.CreateDocument
+import androidx.preference.Preference
 import androidx.preference.PreferenceFragmentCompat
 import com.stevesoltys.seedvault.R
 import com.stevesoltys.seedvault.permitDiskReads
+import org.koin.androidx.viewmodel.ext.android.sharedViewModel
 
 class ExpertSettingsFragment : PreferenceFragmentCompat() {
+
+    private val viewModel: SettingsViewModel by sharedViewModel()
+    // TODO set mimeType when upgrading androidx lib
+    private val createFileLauncher = registerForActivityResult(CreateDocument()) { uri ->
+        viewModel.onLogcatUriReceived(uri)
+    }
+
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
         permitDiskReads {
             setPreferencesFromResource(R.xml.settings_expert, rootKey)
         }
+        findPreference<Preference>("logcat")?.setOnPreferenceClickListener {
+            createFileLauncher.launch("seedvault-log.txt")
```

Comment:
Let's add some more info here, perhaps version and timestamp, so one can have multiple logs.